iio/iio.c

```diff
@@ -954,9 +963,19 @@ static int iio_set_trigger(struct iiod_ctx *ctx, const char *device,
 			   const char *trigger, uint32_t len)
 {
 	struct iio_dev_priv	*dev;
+	struct iio_trig_priv	*trig;
 	uint32_t i;
+	struct iio_desc *desc = ctx->instance;
```

Comment:
maybe `ctx` could be NULL (not sure, double-check)

Reply:
the ctx cannot be null with the current implementation...it comes from iiod: struct iiod_ctx ctx = IIOD_CTX(desc, conn);
also, it is never checked in iio.c

Reply:
ok, fair enough, that's why i asked